src/Thumbnail/LiipImagineThumbnail.php

```diff
@@ -30,7 +32,11 @@ public function generatePublicUrl(MediaProviderInterface $provider, MediaInterfa
     {
         $path = $provider->getReferenceImage($media);
 
-        if (MediaProviderInterface::FORMAT_ADMIN === $format || MediaProviderInterface::FORMAT_REFERENCE === $format) {
+        if (
+            MediaProviderInterface::FORMAT_ADMIN === $format ||
+            MediaProviderInterface::FORMAT_REFERENCE === $format ||
+            \in_array(strtolower(pathinfo((string) $media->getName(), \PATHINFO_EXTENSION)), self::EXCLUDED_FORMATS, true)
```

Comment:
```suggestion
           null !== $media->getName() &&  \in_array(strtolower(pathinfo($media->getName(), \PATHINFO_EXTENSION)), self::EXCLUDED_FORMATS, true)
```

and maybe creating a variable for the media name before the if.